lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -151,8 +152,50 @@ def which(cmd):
     return None
 
 
-class MPI(object):
+class RunFileLoopMPI(RunFile, RunLoop):
+    """
+    Combination of RunFile and RunLoop to support output to file,
+    while also checking whether any output has been produced after a specified amount of time.
+    """
+    def __init__(self, cmd, **kwargs):
+        """
+        handle initialisation: get filename and output timeout from arguments
+        """
+        self.filename = kwargs.get('filename', None)
+        self.output_timeout = kwargs.pop('output_timeout', None)
```

Comment:
please make sure the value here makes sense, i.e. can be parsed as an `int`; if not, raise a `ValueError`